### DISCUSSION TITLE
How are empty rows treated in sparse csc_matrix?

### DISCUSSION THREAD

**Discussion by Levi-Armstrong** [osqp/osqp#547] [General]
I am leveraging OSQP for manipulator motion planning where collision constraints are defined for link pair collision objects. The application uses a contact threshold so the number of collision pairs is dynamic which results in empty rows in the csc_matrix. I plan to prune the rows out but wanted to check if this is already handled internally.

**Comment by goulart-paul** [comment]
They are not filtered out internally - the solver will just take whatever csc_matrix you give it.  The only constraints is that it should be in the standard format, i.e. entries within each column should appear in order of their row numbers (note most libraries / languages will provide it this way automatically).   

I think the solver will also still work even if you have a completely empty row, or a row with non-empty sparse entries but with values set to zero.   No need to pre-filter.

**Comment by Levi-Armstrong** [comment]
Thank you for the quick response. I will try not filtering and see if I have any issues. If I run into any issues I will post them.
